From d15fda2100893c0c474c6cc14670a5db7fa4cc8c Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 21 Apr 2021 17:17:05 +0300
Subject: [PATCH] [SW-26273] fit: convert security error

The security error should be follow the spec's
sec error, for that a convert function was added to
translate the internal/RAS presentation of security error
to the way the proto status needs it.

Change-Id: I745dfa0373c1d959a3dd01c661bc5548cd6d0bfe
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/preboot/fit.c b/preboot/fit.c
index 89caa49..5741953 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -906,6 +906,27 @@
 	i2csi_set_boot_err(boot_err, false);
 }
 
+static int fwupd_sec_convert_min_error(enum fit_sec_failure err)
+{
+	int serr;
+
+	switch (err) {
+	case FIT_SEC_KEY_REVOKED:
+		serr = HL_SEC_ERR_KEY_REVOKED;
+		break;
+	case FIT_SEC_SVN_INVALID:
+		serr = HL_SEC_ERR_SVN_INVALID;
+		break;
+	case FIT_SEC_SIGN_FAILED:
+		serr = HL_SEC_ERR_SIGN_FAILED;
+		break;
+	default:
+		serr = FIT_SEC_NONE;
+	};
+
+	return serr;
+}
+
 int fit_load_images(void)
 {
 	enum fit_sec_failure sec_res = FIT_SEC_NONE;
@@ -933,7 +954,8 @@
 		break;
 	case FIT_LOAD_SEC_FAILED:
 		cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_FAIL, true);
-		preboot_error_inform(HL_FW_ERR_SECURITY, sec_res);
+		preboot_error_inform(HL_FW_ERR_SECURITY,
+				     fwupd_sec_convert_min_error(sec_res));
 		preboot_puts("FIT security validation failed, Retrying...\n");
 		break;
 	}
